src/components/CurrentImage/index.jsx

```diff
@@ -0,0 +1,128 @@
+    // equal to the random number, then pushes that element to the end of newTilesArray.
+    for (let i = 0; i < newTilesArray.length; i++) {
+      let random = Math.floor(Math.random() * (newTilesArray.length - 1));
+      let randomItemArray = newTilesArray.splice(random, 1);
+      newTilesArray.push(randomItemArray[0]);
+    }
+    // Set the randomized array as generated tiles
+    setGeneratedTiles(newTilesArray);
+  };
+
+  useEffect(() => {
+    tilesGenerator();
+  }, []);
+
+  useEffect(() => {
+    checkSolution();
+  }, [generatedTiles]);
+
+  const handleSelect = (currId) => {
+    // if selectedTile's value is not equal to null (meaning there a tile seleted already)
+    // then when another tile is clicked swap both tiles' places
+    // else set the selectedTile to the id of the clicked tile
+    if (selectedTile !== null) {
+      const prevId = selectedTile;
+      swapTiles(prevId, currId);
+    } else {
+      setSelectedTile(currId);
+      const tileIndex = generatedTiles.findIndex((tile) => tile.id === currId);
+      // set the select value of the tile to true
+      generatedTiles[tileIndex] = {
+        ...generatedTiles[tileIndex],
+        selected: true,
+      };
+    }
+  };
+
+  const swapTiles = (id0, id1) => {
+    let newTiles = [...generatedTiles];
+    let index0 = generatedTiles.findIndex((tile) => tile.id === id0);
+    let index1 = generatedTiles.findIndex((tile) => tile.id === id1);
+    let tile0 = { ...generatedTiles[index0] };
+    let tile1 = { ...generatedTiles[index1] };
+    newTiles[index0] = tile1;
+    newTiles[index1] = tile0;
+    newTiles.forEach((tile) => (tile.selected = false));
+    setGeneratedTiles(newTiles);
+    setSelectedTile(null);
+  };
+
+  const checkSolution = () => {
+    // tiles are saved in an array indexed from 0 to n (6) and displayed in that order
+    // first condition checks if tile's id is equal to its index in the array and returns false/true
+    // since initial acc value is true, second condition checks if prev value is true and keeps its value
+    const isCorrect = generatedTiles.reduce((acc, curr, index) => {
+      if (curr.id !== index) {
+        return false;
+      } else if (acc === true) {
+        return acc;
+      }
+    }, true);
+    setIsSolved(isCorrect);
+  };
+
+  return (
+    <Col className="currentImage">
+      {!isSolved ? (
+        generatedTiles.map((tile) => {
+          return (
+            <Row
+              key={"tileNumber-" + tile.id}
+              className={
+                tile.selected
+                  ? "selectedTile currentImage__tile"
+                  : "currentImage__tile"
+              }
+              onClick={() => handleSelect(tile.id)}
+            >
+              <img
+                className="currentImage__puzzledImage"
+                src={puzzledImage}
+                alt="tile from the image"
+                style={{ top: `-${tile.top}%` }}
```

Comment:
this is really weird...shouldn't each img be on a new row and simply stack on top of each other? i don't think this is necessary. but it should definitely be taken around, usually you don't pass CSS information around like this.

i would play with the CSS here.
```
<Row><img></Row>
<Row><img></Row>
<Row><img></Row>
<Row><img></Row>
<Row><img></Row>
```

by changing the css of Row, or using display:block on img, or surrounding the img in divs, you might be able to fix this. or add margins on rows.

Reply:
The way i'm doing this is i'm placing the image inside a div and making the image 6 times bigger than the div then displaying the part that i want to display of that image. I think what you are saying is I should create a new div inside the `<Row>` and then add the CSS to that div instead of add it to the img. please correct me if i misunderstood